src/common/models/retention.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2019 Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package models
+
+import (
+	"time"
+)
+
+// TagRecord represents all pertinent metadata about a tag
+type TagRecord struct {
+	Project    *Project
+	Repository *RepoRecord
+
+	Name       string
+	CreatedAt  time.Time
+	LastPullAt time.Time
+	Labels     []*Label
+}
+
+// Filter is a tag filter in a Retention Policy Filter Chain
+type Filter interface {
```

Comment:
I don't think the interface should be defined in the common model package. You can move it to your own `retention` package.